collections-arrays/ArrayPool/ArrayPool/Program.cs

```diff
@@ -0,0 +1,17 @@
+﻿using static ArrayPool.Methods;
+
+var pool = GetDefaultArrayPool<int>();
+
+var arraySize = 10;
+
+var array = pool.Rent(arraySize);
+
+for (var i = 0; i < arraySize; i++)
+    array[i] = i * 2;
+
+Console.WriteLine("Array elements:");
+
+foreach (var value in array)
+    Console.Write(value + " ");
```

Comment:
This code may not be doing what we think it is doing. The array returned from the ArrayPool when we Rent an array it is only guaranteed that it is at least the size we asked for, but may be bigger. So this foreach could be printing out junk. Here is a dotnet fiddle showing, that in this case, the request for an array of length 10, returns an array of length 16.
https://dotnetfiddle.net/g0jyS2

Reply:
I have explicitly mentioned in the article about the size of the array when we Rent one from the pool. In this specific case, yes it prints the total array. I have written it on purpose to show that when we rent an array from the pool, we get an array of at least the size we request.
Pool returns us the first available array with at least the size we want.

Reply:
@KonstantinosP7 - Got it. I hadn't taken a look at the article text, just commented that it may not be doing what we think, but obviously, it is being used as an example to show the fact that the array may be longer.

Reply:
@KonstantinosP7 - I got to thinking about this more, and this is just my opinion and a thought to consider, but I was wondering if maybe it would be better for us to show the fact that the returned array will be at least as large as we ask and potentially larger, via a simply Console.WriteLine of the requested length and the returned array length, rather than using an example with code that is incorrect or potentially dangerous. This way the potential confusion is avoided. Again, just a thought.

Reply:
@jshergal - I agree with you. Instead of printing all the elements of the rented array, I printed the size instead. You will see that in the PR soon. Thanks!